### PR TITLE
Add units tests for client request handlers

### PR DESCRIPTION
**Pull request by delventhalz**

### CODE REVIEW

File: validator/tests/unit3/test_client_request_handlers/tests.py [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    def _initialize(self, handler, request_proto, response_proto,
+                    store=None, roots=None):
+        self._identity = '1234567'
+        self._handler = handler
+        self._status = response_proto
+        self._request_proto = request_proto
+        self._store = store
+        self._roots = roots
+
+    def _make_request(self, **kwargs):
```
Comment (vaporos) [inline validator/tests/unit3/test_client_request_handlers/tests.py]: These are used by the subclass, so probably shouldn't be prefixed with _.

File: validator/tests/unit3/test_client_request_handlers/tests.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+    def _serialize(self, **kwargs):
+        request = self._request_proto(**kwargs)
+        return request.SerializeToString()
+
+    def _handle(self, request):
+        result = self._handler.handle(self._identity, request)
+        return result.message_out
+
+    def _make_bad_request(self, **kwargs):
```
Comment (vaporos) [inline validator/tests/unit3/test_client_request_handlers/tests.py]: What is the significance here that differentiates a bad request from a normal request?

File: validator/tests/unit3/test_client_request_handlers/tests.py [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+            roots=roots)
+
+    def test_state_list_request(self):
+        """Verifies requests for data lists without parameters work properly.
+        Checks status, head_id, and length and type of leaves list.
+        """
+        response = self._make_request()
+
+        self.assertEqual(self._status.OK, response.status)
+        self.assertEqual('2', response.head_id)
```
Comment (vaporos) [inline validator/tests/unit3/test_client_request_handlers/tests.py]: Additional comments in the API doc explaining the magic values here ('2' and 3) would help (why should it be '2' and 3 at this point?).  (Comment really applies to several of the tests here that test 'magic' values.)

File: validator/tests/unit3/test_client_request_handlers/tests.py [hunk 4]
```diff
@@ -0,0 +1,410 @@
+# Copyright 2016 Intel Corporation
```
Comment (vaporos) [inline validator/tests/unit3/test_client_request_handlers/tests.py]: Should be 2017

File: validator/tests/unit3/test_client_request_handlers/tests.py (hunk 2 above)
Comment (delventhalz) [inline validator/tests/unit3/test_client_request_handlers/tests.py]: It truncates the protobuf.

**Pull request merged**